price_parser/parser.py

```diff
@@ -36,14 +36,15 @@ def fromstring(cls, price: Optional[str],
         ``price`` string, it could be **preferred** over a value extracted
         from ``currency_hint`` string.
         """
-        amount_text = extract_price_text(price) if price is not None else None
+        currency, source = _extract_currency_symbol(price, currency_hint)
+        amount_text = extract_price_text(price, currency if source == price else None) if price is not None else None
```

Comment:
could you please split it into several lines - maybe just use regular `if` statement? We're trying to use 79 line length limit here, unless there is a reason not to.